### ISSUE TITLE
nls error

### ISSUE THREAD

**Issue by quantixed** [quantixed/TrackMateR#2]
Thanks ! 
Now it works for the example dataset, but when trying this datased (attached file, xml from trackmate) I get this error when using the report command:

Error in nls(hdata$countsCum ~ (1 - D2 * exp(-hdata$mid^2/(4 * D1 * timeRes)) -  : 
  singular gradient

[xml.zip](https://github.com/quantixed/TrackMateR/files/9526754/xml.zip)

_Originally posted by @fadelvalle in https://github.com/quantixed/TrackMateR/issues/1#issuecomment-1240649539_

**Comment by quantixed** [comment]
I get a different but related error:
`Error in nls(hdata$countsCum ~ (1 - D2 * exp(-hdata$mid^2/(4 * D1 * timeRes)) -  : 
  step factor 0.000488281 reduced below 'minFactor' of 0.000976562`

The fitting step here and jump distance needs to be improved or at least any errors should dealt with better.

**Comment by fadelvalle** [comment]
Yes, it works that way. When I try to plot the result with  plot_tm_MSD() using as input msdcalc<-calculateMSD(tmDF) it just works with the first first element : plot_tm_MSD(msdcalc[[1]]) and gives this graph:
![image](https://user-images.githubusercontent.com/27208032/189143562-0a292ecb-7efd-4252-bc2d-91751856d2e1.png)

when I try the command plot_tm_MSD(msdcalc) it gives this error:
> plot_tm_MSD(msdcalc)
Error in df[1:4, ] : incorrect number of dimensions

**Comment by quantixed** [comment]
That's not a bug. But I can change it so that if a user feeds in the wrong this it's handled better.

**Comment by fadelvalle** [comment]
thanks again for all this fast and very useful feedback !

**Comment by quantixed** [comment]
@fadelvalle here is a script to make a report from a TrackMate XML file like the one you uploaded. I thought I would send you this so you can look at your data without waiting for me to fix the nls error.
![Rplot03](https://user-images.githubusercontent.com/13585138/189161974-f63c6800-2615-46d4-b8f0-56921df591bf.png)
[testScript.R.zip](https://github.com/quantixed/TrackMateR/files/9528199/testScript.R.zip)

**Comment by fadelvalle** [comment]
Thank you so much. I will take a look into the script so I can compare data

**Comment by quantixed** [comment]
I have fixed this problem which was caused by the initial guesses for jump distance fitting being too rigid. Now `fittingJD()` takes a better guess at the initial parameters (using the average jump distance, the time resolution and binning of the distance data to make the guess). Second, if this fails, the error does not prevent the report from being made. Third, users can now fine tune the fitting either if they want to or if they get an error from the automatic guessing procedure.
https://github.com/quantixed/TrackMateR/commit/2681bd12a6f2e58fe6924e132706c88644758f6f
